common/domain/replicationTaskExecutor.go

```diff
@@ -262,6 +267,7 @@ func (h *domainReplicationTaskExecutorImpl) handleDomainUpdateReplicationTask(ct
 	}
 
 	if !recordUpdated {
+		h.logger.Warn("Domain update failed, record not updated")
```

Comment:
i still don't think we should add this log

Reply:
What is your concern? I don't see how this is a noisy log

Reply:
The message is misleading and it's asking people to investigate a non-issue.

Reply:
Maybe my mental model is wrong, under what conditions should a conditional update normally occur in this codepath? I get domain update queue replays, but that seems fairly rare

Reply:
I don't know the details, but it probably could occur when merging dlq.